.github/workflows/nix.yml

```diff
@@ -0,0 +1,18 @@
+name: Nix
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    branches: [ "main" ]
+
+jobs:
+  nix-flake:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@08c6903cd8c0fde910a37f88322edcfb5dd907a8 # v5.0.0
+    - uses: cachix/install-nix-action@v27
+      with:
+        nix_path: nixpkgs=channel:nixos-unstable
+    - run: nix flake check
+    - run: nix build
```

Comment:
Speaking from experience, Nix builds can take a loooooong time. We should probably use something like https://github.com/DeterminateSystems/magic-nix-cache-action (which uses the GH actions cache) or https://github.com/cachix/cachix-action if we want to set up a Cachix for wassette

Reply:
I was going to open an issue to request a Cachix key added to the repository.

Magic cache is great, but it's also very slow because it uses the actions cache.

Reply:
Sweet. I'll set that up now actually.

Reply:
@rawkode I've added `CACHIX_AUTH_TOKEN` and the cache name is wassette

Reply:
Thank you. Pushed.